LogMonitor/src/LogMonitor/PipeMonitor.cpp

```diff
@@ -16,13 +15,39 @@ DWORD HandlePipeStream(HANDLE hPipe) {
     char chBuf[BUFSIZE];
     bool bSuccess = false;
 
+    // Loop until done reading
     while (1) {
-        bSuccess = ReadFile(hPipe, chBuf, BUFSIZE * sizeof(char), &dwRead, NULL);
+        // Read client requests from the pipe. only allows messages up to BUFSIZE characters in length.
+        bSuccess = ReadFile(
+            hPipe,      // handle to pipe 
+            chBuf,      // buffer to receive data
+            BUFSIZE * sizeof(char), // size of buffer
+            &dwRead,                // number of bytes read
+            NULL);                  // not overlapped I/O 
 
         DWORD readFileError = GetLastError();
         if (!bSuccess || dwRead == 0)
         {
-            logWriter.WriteConsoleLog(std::to_wstring(readFileError));
+            switch (readFileError)
+            {
+            case ERROR_BROKEN_PIPE:

```

Comment:
Should we log an error or silently handle this? Can we distinguish an absolute failure to connect from a client disconnect? In the image below, when ping exits, we throw this error instead of allowing a silent disconnect

![image](https://user-images.githubusercontent.com/50120072/191679149-16e03298-9322-4eaa-b185-2401585874e6.png)

Reply:
same case for echo "Your message here" > \\.\pipe\logMonitor

Reply:
Just to clarify with the example above identify that we our client comes from ping in the first case and echo in the second.

Reply:
Also, how about having a message like this:

_**Control-C                                                                                                               
[2022-09-22T22:32:45.000Z][LOGMONITOR] INFO: Client disconnected.**_ 


To silently handle the case where the client disconnected

Reply:
@bob This works - trying to see whether we can provide more info around which client disconnected.

Reply:
Testing this: GetNamedPipeClientProcessId

Reply:
@iankingori at this point we have the STDOUT handle of the parent in hParentStdOut variable so this can help to get more info on the client that disconnected.